xilem/src/view/button.rs

```diff
@@ -89,10 +94,40 @@ pub struct Button<F> {
     // type of `Label` even though it currently does not do so.
     label: Label,
     callback: F,
+    properties: (
+        Option<Background>,
+        Option<BorderColor>,
+        Option<BorderWidth>,
+        Option<BoxShadow>,
+        Option<CornerRadius>,
+        Option<Padding>,
+    ),
 }
 
 const LABEL_VIEW_ID: ViewId = ViewId::new(0);
 
+impl<F> Style for Button<F> {
+    type Props = (
+        Option<Background>,
+        Option<BorderColor>,
+        Option<BorderWidth>,
+        Option<BoxShadow>,
+        Option<CornerRadius>,
+        Option<Padding>,
+    );
+
+    fn properties(&mut self) -> &mut Self::Props {
+        &mut self.properties
+    }
+}
+
+impl<F> HasProperty<Background> for Button<F> {}
+impl<F> HasProperty<BorderColor> for Button<F> {}
+impl<F> HasProperty<BorderWidth> for Button<F> {}
+impl<F> HasProperty<BoxShadow> for Button<F> {}
+impl<F> HasProperty<CornerRadius> for Button<F> {}
+impl<F> HasProperty<Padding> for Button<F> {}
```

Comment:
I do worry that it would be very easy to miss a property here. My suggested alternative does resolve that concern, if the final impl were macro based.